uint/src/uint.rs

```diff
@@ -50,6 +50,26 @@ macro_rules! impl_map_from {
 	}
 }
 
+#[macro_export]
+#[doc(hidden)]
+macro_rules! impl_try_from_for_primitive {
+	($from:ident, $to:ty) => {
+		impl $crate::core_::convert::TryFrom<$from> for $to {
+			type Error = &'static str;
+
+			#[inline]
+			fn try_from(u: $from) -> Result<$to, &'static str> {
+				let $from(arr) = u;
+				if !u.fits_word() || arr[0] > <$to>::max_value() as u64 {
+					Err("integer overflow when casting")
```

Comment:
Can we include the types involved in the error?

Reply:
Done :]